### PR TITLE
Move ResourceAgent to wire-types and make check_ha to async/await

### PR DESCRIPTION
**Pull request by utopiabound**
Add very basic test.

Fixes #1288 #1232 

Signed-off-by: Nathaniel Clark <nclark@whamcloud.com>

---
This change is [Reviewable](https://reviewable.io/reviews/whamcloud/integrated-manager-for-lustre/1442)

### CODE REVIEW

File: iml-wire-types/wire-types.rs [hunk 1]
```diff
@@ -1998,6 +1998,42 @@ pub struct ComponentState<T: Default> {
...
+            ocftype,
+        }
+    }
+}
+
+impl std::fmt::Display for ResourceAgentType {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match &self.provider {
+            Some(provider) => write!(f, "{}:{}:{}", self.standard, provider, self.ocftype),
+            None => write!(f, "{}:{}", self.standard, self.ocftype),
```
Comment (ip1981) [inline iml-wire-types/wire-types.rs]: Out of curiosity: where is this used? I think looking at `foo:bar` you can't decide whether `bar` is provider or ocftype without looking ahead.

File: iml-agent/src/action_plugins/check_ha.rs [hunk 2]
```diff
@@ -4,90 +4,137 @@
...
-                                .map(|p| AgentInfo::create(p, None)),
-                        )
-                        .collect(),
-                ),
-            }
-        })
-        .err_into()
+    #[test]
+    fn test_good() {
+        let testxml = r#"<resources>
```
Comment (ip1981) [inline iml-agent/src/action_plugins/check_ha.rs]: Possible to put in a separate file?

BTW, it seems missing an XML declaration - `<?xml version="1.0" ... ?>`

File: iml-wire-types/wire-types.rs (hunk 1 above)
Comment (utopiabound) [reply iml-wire-types/wire-types.rs]: That display format is standard for pacemaker output.

File: iml-agent/src/action_plugins/check_ha.rs (hunk 2 above)
Comment (utopiabound) [reply iml-agent/src/action_plugins/check_ha.rs]: It's not missing the declaration.  This is testing the output of `cibadmin --query --xpath //resources`

File: iml-wire-types/wire-types.rs [hunk 3]
```diff
@@ -2059,6 +2059,42 @@ pub struct ComponentState<T: Default> {
...
+    pub ocftype: String,          // e.g. Lustre, ZFS
+}
+
+impl ResourceAgentType {
+    pub const fn new(standard: String, provider: Option<String>, ocftype: String) -> Self {
+        ResourceAgentType {
+            standard,
+            provider,
+            ocftype,
+        }
```
Comment (nlinker) [inline iml-wire-types/wire-types.rs]: I think
```
        Self {
              standard,
              provider,
              ocftype,
          }
```
is nicer.

**Pull request closed without merging**